lib/components/TextInput.jsx

```diff
@@ -0,0 +1,203 @@
+import React, { Component, PropTypes } from 'react';
+import InputLabel from './InputLabel';
+
+/**
+ * Class representing a label and a input of type="text"
+ *
+ * Required props:
+ * - label: Sets the text for the input's label
+ * - id: sets the id attribute for the input and the <label for... attribute
+ *
+ * Optional props:
+ * - value: string. If present pre-populates the input with the given string.
+ * - required: bool, defaults to false. Adds required label, required attribute and aria-required='true'
+ * - validators: array of Validator objects. If present, field get error state onBlur if the value does not match the given test
+ * - spellCheck: true or false, defaults to false. If true, enables browser autocorrection while typing
+ * - errorMessage: string. If present triggers the error state and displays the error message
+ * - isValid: bool. If true, sets isValid state - i.e. green border
+ * - allowedChars: Validator object. If present, user can only type characters that match the given test
```

Comment:
Yes, the thought was allowedChars would evaluated as the user types to prevent an "illegal" character from being entered ... i.e. prevent letters from being entered into a ZIP Code field. It's very possible that 508 would hate that very much. If it is kosher and we want to use it, it would have to be used along with the validators ... at least how it is implemented now. Continuing on the ZIP Code example, just because there's no letters or punctuation in the field doesn't mean it's a valid ZIP Code